### PR TITLE
docs: add a migration guide for the format string changes

### LINKED ISSUE #1404: Create migration guide for the format string release
#### Is your feature request related to a problem? Please describe.

As part of the format strings release there are a few breaking changes being made to the config file we will need to provide a migration guide to help our users with the migration.

#### Describe the solution you'd like

> ... Here are the breaking changes I've come about:
> 
> - `prompt_order` -> `format`
> - `add_newline` -> `format`
> - `SegmentConfig` related
```toml
symbol = {value = "S ", style = "green"} # Before
symbol = "[S ](green)" # After
```
> - A few modules (`character`, `directory`, `memory` and `time`)

_Originally posted by @heyrict in https://github.com/starship/starship/pull/1078#issuecomment-650134547_

### PR DESCRIPTION
**Pull request by matchai**
#### Description

Add a migration guide for the format string changes introduced in v0.45.0.

#### Motivation and Context

Closes #1404

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

**Comment by andytom** [comment]
Hey @matchai I was trying to check this on the preview site that netlify generated, https://deploy-preview-1624--starship-rs.netlify.app/, but I couldn't find the guide am I missing something?

**Comment by matchai** [comment]
We can edit the config to add the link to the sidebar. For the time being you can access it here: https://deploy-preview-1624--starship-rs.netlify.app/migrating-to-0.45.0/

File: docs/migrating-to-0.45.0/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+Starship v0.45.0 will instead be accepting a `format` value, allowing for further
+customization of how modules are rendered. Instead of defining a prefix and suffix
+for the context-based variables, the variables can now be substituted from within
+a format string, which represents the module's output.
+
+**Example pre-v0.45.0 configuration**
+
+```toml
+[time]
```
Comment (andytom) [inline docs/migrating-to-0.45.0/README.md:69]: I think this should be

```suggestion
[cmd_duration]
```

File: docs/migrating-to-0.45.0/README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+prefix = "took "
+```
+
+**Example v0.45.0 configuration**
+
+```toml
+[time]
```
Comment (andytom) [inline docs/migrating-to-0.45.0/README.md:76]: And here

```suggestion
[cmd_duration]
```

**Review by andytom — APPROVED** [review]
Have added a little suggestion but this LGTM. Do we want to also include any breaking changes to the individual modules here or should that be somewhere else?

**Comment by matchai** [comment]
I think this may be the best place to put per-module breaking changes.
I'll try to update this PR to include all of those 👍

**Comment by matchai** [comment]
I updated the PR to include guidelines for the changed modules.

File: docs/migrating-to-0.45.0/README.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+```toml
+[cmd_duration]
+# $duration – The command duration (e.g. "15s")
+# $style    – The default style of the module (e.g. "bold yellow")
+format = "took [$duration]($style)"
+```
+
+### Affected Modules
+
```
Comment (andytom) [inline docs/migrating-to-0.45.0/README.md:83]: Character also has breaking changes

```suggestion

#### Character

| Deprecated Property | Replacement |
| ------------------- | ----------- |
| `symbol` | `success_symbol` |
| `use_symbol_for_status` | `success_symbol`, `error_symbol` |
| `style_success` | `success_symbol` |
| `style_failure` | `error_symbol` |

```diff
[character]
-- symbol = "➜"
-- error_symbol = "✗"
-- use_symbol_for_status = true
++ success_symbol = "[➜](bold green) "                                                                                     
++ error_symbol = "[✗](bold red) "
```

```

**Review by andytom — COMMENTED** [review]
I've added suggestions for the few module changes I know about but I think there might be a few more that I've missed.

File: docs/migrating-to-0.45.0/README.md (hunk 3 above)
Comment (matchai) [reply docs/migrating-to-0.45.0/README.md:83]: I'm pretty sure that the new character configuration is the following:

```toml
success_symbol = "[❯](bold green)"
error_symbol = "[❯](bold red)"
```

Comment (andytom) [reply docs/migrating-to-0.45.0/README.md:83]: Ah yep you are right, I just copied my local config.

**Pull request merged**